cypher-shell/src/main/java/org/neo4j/shell/CypherShell.java

```diff
@@ -12,20 +12,22 @@
 import org.neo4j.shell.state.BoltStateHandler;
 
 import javax.annotation.Nonnull;
+import java.util.AbstractMap;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 /**
  * A possibly interactive shell for evaluating cypher statements.
  */
 public class CypherShell implements StatementExecuter, Connector, TransactionHandler, VariableHolder {
     // Final space to catch newline
     protected static final Pattern cmdNamePattern = Pattern.compile("^\\s*(?<name>[^\\s]+)\\b(?<args>.*)\\s*$");
-    protected final Map<String, Object> queryParams = new HashMap<>();
+    protected final Map<String, AbstractMap.SimpleEntry<String, Object>> queryParams = new HashMap<>();
```

Comment:
I think it is better to create a simple class with `String` and `Object` fields instead of using very verbose `AbstractMap.SimpleEntry<String, Object>` everywhere.

Reply:
You'd create temporal and spatial parameters using functions similar to the usage in queries ie: `point({x:1, y:2})` 
About printing Spatial and Temporal types we based on @apcj excellent direction decided to use `toString`. I believe he'll be taking this discussion forward.

Reply:
I'm very much against using `toString()`. Java driver provides no explicit guarantees on the `toString()` format. Cypher shell has it's own requirements on how parameters should be formatted and thus should take care of the formatting. Please see the following two links for reasons why `toString()` is not the best choice: [one](https://stackoverflow.com/questions/19911290/java-tostring-for-debugging-or-actual-logical-use) & [two](https://www.javaworld.com/article/2073619/core-java/java-tostring---considerations.html)